elyra/pipeline/tests/test_handlers.py

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright 2018-2021 Elyra Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+import pytest
+import tornado
+
+from tornado.httpclient import HTTPClientError
+
+try:
+    import importlib.resources as pkg_resources
+except ImportError:
+    # Try backported to PY<37 `importlib_resources`.
+    import importlib_resources as pkg_resources
+
+from .. import resources
+
+
+# Set Elyra server extension as enabled (overriding server_config fixture from jupyter_server)
+@pytest.fixture
+def jp_server_config():
+    return {
+        "ServerApp": {
+            "jpserver_extensions": {
+                "elyra": True
+            }
+        }
+    }
+
+
+async def test_invalid_config_resource(jp_fetch):
+    with pytest.raises(HTTPClientError) as e:
+        await jp_fetch('elyra', 'pipeline', 'config', 'invalid', method='GET')
+
+    assert expected_http_error(e, 400)
+
+
+async def test_get_palette_config(jp_fetch):
+    # Ensure all valid metadata can be found
+    response = await jp_fetch('elyra', 'pipeline', 'config', 'palette')
+    assert response.code == 200
+    payload = json.loads(response.body.decode())
+    palette = json.loads(pkg_resources.read_text(resources, 'palette.json'))
+    assert payload == palette
+
+
+async def test_get_properties_config(jp_fetch):
+    # Ensure all valid metadata can be found
+    response = await jp_fetch('elyra', 'pipeline', 'config', 'properties')
+    assert response.code == 200
+    payload = json.loads(response.body.decode())
+    properties = json.loads(pkg_resources.read_text(resources, 'properties.json'))
+    assert payload == properties
+
+
+def expected_http_error(error, expected_code, expected_message=None):
+    """Check that the error matches the expected output error."""
+    e = error.value
+    if isinstance(e, tornado.web.HTTPError):
+        if expected_code != e.status_code:
+            return False
+        if expected_message is not None and expected_message != str(e):
```

Comment:
I think a strict equality test will be difficult (and see there are not calls that actually validate the message text).  I would suggest using `expected_message in str(e)` instead.  Then the dev only needs to provide the applicable segment of the message while allowing for variations (like temp filenames for example) introduced by the test framework.

Reply:
so, this utility function is a copy from metadata, which is a copy from jupyter_server. Can we get rid of these local copies and use the one from jupyter_server? should we expose those from there?

Reply:
Yeah, this rings a bell now.   This function was used directly in `metadata/tests/test_utils.py` because, at that time, the server tests were external to the package.  However, since then, they've been moved back into the package so that users could run tests after `pip install`.  So, I agree, just use the one from server now that it's available.  Do you mind removing the function from the metadata/tests/test_utils and updating the import statement in metadata/tests/test_handlers?

I see none of the metadata tests use the message parameter, but there are some jupyter_server tests that use specific substitutions for the message parameter to allow equality to work - which I'm sure requires debugging to get that right, but that's fine (especially since none of our tests use it).

Reply:
removed the duplication and we are now importing it from jupyter-server